src/wrappers/themis/wasm/src/secure_cell_context_imprint.ts

```diff
@@ -0,0 +1,260 @@
+// Copyright (c) 2019 Cossack Labs Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file
+ * Themis Secure Cell (Context Imprint mode).
+ */
+
+import themisContext from "./context";
+import { ThemisError, ThemisErrorCode } from "./themis_error";
+import {
+  coerceToBytes,
+  heapFree,
+  heapGetArray,
+  heapPutArray,
+  heapAlloc,
+} from "./utils";
+
+const cryptosystem_name = "SecureCellContextImprint";
+
+export class SecureCellContextImprint {
+  private masterKey: Uint8Array;
+
+  constructor(masterKey: Uint8Array) {
+    masterKey = coerceToBytes(masterKey);
+    if (masterKey.length == 0) {
+      throw new ThemisError(
+        cryptosystem_name,
+        ThemisErrorCode.INVALID_PARAMETER,
+        "master key must be not empty"
+      );
+    }
+    this.masterKey = masterKey;
+  }
+
+  /**
+   * Makes a new Secure Cell in Context Imprint mode with given master key.
+   *
+   * @param masterKey     non-empty array of master key bytes (Buffer or Uint8Array)
+   *
+   * @returns a new instance of SecureCellContextImprint.
+   *
+   * @throws TypeError if the master key is not a byte buffer.
+   * @throws ThemisError if the master key is empty.
+   */
+  static withKey(masterKey: Uint8Array) {
+    return new SecureCellContextImprint(masterKey);
+  }
+
+  encrypt(message: Uint8Array, context: Uint8Array) {
+    message = coerceToBytes(message);
```

Comment:
In a future PR I would like to remove all usages of `coerceToBytes` and demand a Uint8Array to be passed to the library. That would simplify the code in many places.